apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications.js

```diff
@@ -0,0 +1,299 @@
+        where: {
+            date_gte: startWindowDate,
+            date_lte: endWindowDate,
+            meter: {
+                id_in: meterIds,
+            },
+        },
+    })
+}
+
+const readOrganizations = async ({ context, meters }) => {
+    // first step is get unique organizations ids
+    const organizationIds = uniq(meters.map(meter => meter.organization.id))
+
+    // load all pages for entity
+    return await loadListByChunks({
+        context,
+        list: Organization,
+        where: {
+            id_in: organizationIds,
+        },
+    })
+}
+
+const getLocalizedMeterResources = async ({ context, lang }) => {
+    // get context for specific locale
+    const localeDependedContext = {
+        ...context.createContext({ skipAccessControl: true }),
+        req: { locale: lang },
+    }
+
+    // get list of meter resources for certain locale
+    // TODO apply DOMA-3174 mechanic for custom request locale
+    return await loadListByChunks({
+        context,
+        list: MeterResource,
+        where: {},
+    })
+}
+
+const loadAllLocalizedVersionsOfMeterResources = async ({ context }) => {
+    const supportedLanguages = ['en', 'ru']
+    const result = {}
+
+    for (let supportedLanguage of supportedLanguages) {
+        result[supportedLanguage] = await getLocalizedMeterResources({ context, lang: supportedLanguage })
+    }
+
+    return result
+}
+
+const sendSubmitMeterReadingsPushNotifications = async () => {
+    // task implementation algorithm
+    // * Read all meter resources for all supported languages
+    // * Read meters page
+    // * Read meter readings (from 01 day of month to now) for meters above
+    // * Read meter organisation for meters above
+    // * Right join readings/organisation/resources to meters
+    // * Filter meters without readings
+    // * Filter meters without organisation default period
+    // * Join residents to rest of meters
+    // * Send message with specific unique key for set up readings (key is: {meterId}_{residentId}_{now().minusMonth().format(YYYY_MM)})
+    // * Filter meters with expired next verification date
+    // * Send message with specific unique key for expired verification (key is: {meterId}_{residentId}_{now().minusMonth().format(YYYY_MM)})
+    // * Print some stat
+
+    log('Start sending submit meter notifications', '=')
+    // initialize context stuff
+    const { keystone: context } = await getSchemaCtx('Meter')
+    const resourcesLocalizations = await loadAllLocalizedVersionsOfMeterResources({ context })
+
+    // let's load meters page by page
+    const state = {
+        pageSize: 100,
+        offset: 0,
+        hasMore: true,
+        processedMeters: 0,
+        processedPages: 0,
+        metersWithoutReadings: 0,
+        metersWithoutReadingsAndWithResidents: 0,
+        metersWithExpiredVerificationDate: 0,
+        startTime: dayjs(),
+    }
+    log(`Start proceeding meters page by page ${ state.startTime }`, '-')
```

Comment:
it's better not to put variables inside the message. You con't filter by text. Use json messages to simplify filtering

Reply:
And we don't use a ` ` (space) inside `${ }

Reply:
state.startTime - added here only for information purposes. The main goal of that - have some sort of info about how long it is working

Reply:
intent fixed